java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -458,24 +459,25 @@ private Resolution findConstructor(Env env, JavaType site, List<JavaType> argTyp
 
   private Resolution findMethodByStrictThenLooseInvocation(Env env, JavaType callSite, JavaType site, String name, List<JavaType> argTypes, List<JavaType> typeParams) {
     // JLS8 - §5.3 searching by strict invocation, then loose invocation
-    Resolution bestSoFar = findMethod(env, callSite, site, name, argTypes, typeParams, false);
+    Resolution bestSoFar = findMethod(env, callSite, site, name, argTypes, typeParams, false, false);
     // searching for a specific method applicable with fixed arity and loose invocation
-    if (!argTypes.isEmpty() && (bestSoFar.symbol.kind >= JavaSymbol.ERRONEOUS || canFindStrictInvocationWithLooseInvocation(argTypes, bestSoFar))) {
+    if (!argTypes.isEmpty() && bestSoFar.symbol.kind >= JavaSymbol.ERRONEOUS) {
       // retry with loose invocation
-      bestSoFar = findMethod(env, callSite, site, name, argTypes, typeParams, true);
+      bestSoFar = findMethod(env, callSite, site, name, argTypes, typeParams, true, false);
+    }
+    if(bestSoFar.symbol.kind >= JavaSymbol.ERRONEOUS) {
+      // loose invocation and var arity
+      bestSoFar = findMethod(env, callSite, site, name, argTypes, typeParams, true, true);
     }
     return bestSoFar;
   }
 
-  private static boolean canFindStrictInvocationWithLooseInvocation(List<JavaType> argTypes, Resolution bestSoFar) {
-    return ((JavaSymbol.MethodJavaSymbol) bestSoFar.symbol).isVarArgs() && argTypes.stream().anyMatch(t -> t.isPrimitive() || t.isPrimitiveWrapper());
-  }
-
-  private Resolution findMethod(Env env, JavaType callSite, JavaType site, String name, List<JavaType> argTypes, List<JavaType> typeParams, boolean looseInvocation) {
+  private Resolution findMethod(Env env, JavaType callSite, JavaType site, String name, List<JavaType> argTypes, List<JavaType> typeParams,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)